lib/idx/remediate.ts

```diff
@@ -139,32 +148,53 @@ function handleIdxError(e, flow, remediator?) {
   throw e;
 }
 
+function getActionsFromParameters(parameters): string[] {
+  return Object.keys(parameters).map(parameter => actionsTriggeredByParameters[parameter]).filter(Boolean);
+}
+
+function removeActionFromParameters(parameters, action) {
+  const executedActionParameter = Object.keys(actionsTriggeredByParameters).find(
+    parameter => actionsTriggeredByParameters[parameter] === action);
+  return Object.keys(parameters).filter(parameter => parameter !== executedActionParameter)
+  .reduce((newValues, valueName) => {
+    newValues[valueName] = parameters[valueName];
+    return newValues;
+  }, {});
+}
+
 // This function is called recursively until it reaches success or cannot be remediated
 export async function remediate(
   idxResponse: IdxResponse,
   values: RemediationValues,
   options: RunOptions
 ): Promise<RemediationResponse> {
-  const { neededToProceed } = idxResponse;
-  const { actions, flow, flowMonitor } = options;
+  let { neededToProceed } = idxResponse;
+  const { flow, flowMonitor } = options;
+
+  const actions = [
+    ...options.actions || [] as string[],
+    ...getActionsFromParameters(values) || [] as string[]
+  ];
+  let valuesWithoutExecutedAction = values;
```

Comment:
looks like it should be 
`values = removeActionFromParameters(values, action);` 
as execute action does need values, but we want the following remediate calls to be triggered without actionValue.

Reply:
right, action value is getting removed right after action is [executed](https://github.com/okta/okta-auth-js/pull/837/files/79087ed6962d5ff1ac31be5755329ccc9be7f246#diff-39e2797a5978736cacc4da883ee42d837f0c9fa0854b2a75d0fb227388fac3f1R186) - good to remove it before action call?

Reply:
I think so. The purpose for that value is to add a new action to the actions list, then it should be removed right after that operation.